### PR TITLE
added E2E test for header and dashboard

### PR DESCRIPTION
**Pull request by NileshPatel17**
added E2Ee test for #911

### CODE REVIEW

File: apps/sensenet/cypress/integration/dashboard/dashboard.spec.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  before(() => {
+    cy.login()
+    cy.visit(pathWithQueryParams({ path: '/', newParams: { repoUrl } }))
+  })
+  it('Header should contains the url of the current repository.', () => {
+    cy.get('[data-test="sensenet-header"]').contains(repoUrl)
+  })
+
+  it(`should have title`, () => {
+    cy.get('h1').contains(/Welcome to your [a-z0-9]+ project/)
```
Comment (herflis) [inline apps/sensenet/cypress/integration/dashboard/dashboard.spec.ts:15]: please add a `data-test` attribute to the h1 element (as you did with the header) and select the element by this attr

File: apps/sensenet/cypress/integration/dashboard/dashboard.spec.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  })
+
+  it('Subscription section should have the Developer plan text and features list', () => {
+    expect('3 users').to.match(/[0-3] users/)
```
Comment (herflis) [inline apps/sensenet/cypress/integration/dashboard/dashboard.spec.ts:19]: please add a `data-test` attribute to the element (as you did with the header) and select the element by this attribute

File: apps/sensenet/cypress/integration/dashboard/dashboard.spec.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    expect('500 content').to.match(/[0-500] content/)
```
Comment (herflis) [inline apps/sensenet/cypress/integration/dashboard/dashboard.spec.ts:20]: please add a `data-test` attribute to the element (as you did with the header) and select the element by this attribute

File: apps/sensenet/cypress/integration/dashboard/dashboard.spec.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    expect('1 GB storage space').to.match(/[0-1] GB storage space/)
```
Comment (herflis) [inline apps/sensenet/cypress/integration/dashboard/dashboard.spec.ts:21]: please add a `data-test` attribute to the element (as you did with the header) and select the element by this attribute

File: apps/sensenet/cypress/integration/dashboard/dashboard.spec.ts [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  })
+
+  it('Current usage section should have correct usage info.', () => {
+    expect('3 of 3 used').to.match(/[0-3] of 3 used/)
```
Comment (herflis) [inline apps/sensenet/cypress/integration/dashboard/dashboard.spec.ts:25]: please add a `data-test` attribute to the element (as you did with the header) and select the element by this attribute

File: apps/sensenet/cypress/integration/dashboard/dashboard.spec.ts [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    expect('21 of 500 used').to.match(/[0-9] of 500 used/)
```
Comment (herflis) [inline apps/sensenet/cypress/integration/dashboard/dashboard.spec.ts:26]: please add a `data-test` attribute to the element (as you did with the header) and select the element by this attribute

File: apps/sensenet/cypress/integration/dashboard/dashboard.spec.ts [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+    expect('0 of 1 GB used').to.match(/[0-1] of 1 GB used/)
```
Comment (herflis) [inline apps/sensenet/cypress/integration/dashboard/dashboard.spec.ts:27]: please add a `data-test` attribute to the element (as you did with the header) and select the element by this attribute

**Review by herflis — CHANGES_REQUESTED** [review]

**Comment by NileshPatel17** [comment]
All requested changes are done.

File: apps/sensenet/cypress/integration/dashboard/dashboard.spec.ts [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+
+  it('Subscription section should have the Developer plan text and features list', () => {
+    cy.get('[data-test="feature-users"]').contains(/[0-3] users/)
+    cy.get('[data-test="feature-content"]').contains(/[0-500] content/)
+    cy.get('[data-test="feature-storage-space"]').contains(/[0-1] GB storage space/)
+  })
+
+  it('Current usage section should have correct usage info.', () => {
+    cy.get('[data-test="usage-users"]').contains(/[0-3] of 3 used/)
+    cy.get('[data-test="usage-contents"]').contains(/[0-9] of 500 used/)
```
Comment (herflis) [inline apps/sensenet/cypress/integration/dashboard/dashboard.spec.ts:26]:
```suggestion
    cy.get('[data-test="usage-contents"]').contains(/[0-500] of 500 used/)
```

Comment (NileshPatel17) [reply apps/sensenet/cypress/integration/dashboard/dashboard.spec.ts]: Done

File: apps/sensenet/cypress/integration/dashboard/dashboard.spec.ts [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+    cy.get('[data-test="sensenet-header"]').contains(repoUrl)
+  })
+
+  it(`should have title`, () => {
+    cy.get('[data-test="app-header"]').contains(/Welcome to your [a-z0-9]+ project/)
+  })
+
+  it('Subscription section should have the Developer plan text and features list', () => {
+    cy.get('[data-test="feature-users"]').contains(/[0-3] users/)
+    cy.get('[data-test="feature-content"]').contains(/[0-500] content/)
```
Comment (herflis) [inline apps/sensenet/cypress/integration/dashboard/dashboard.spec.ts:20]:
```suggestion
    cy.get('[data-test="feature-content"]').contains(/([0-9]|[1-8][0-9]|9[0-9]|[1-4][0-9]{2}|500) content/)
```

File: apps/sensenet/cypress/integration/dashboard/dashboard.spec.ts [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+    cy.get('[data-test="feature-storage-space"]').contains(/[0-1] GB storage space/)
+  })
+
+  it('Current usage section should have correct usage info.', () => {
+    cy.get('[data-test="usage-users"]').contains(/[0-3] of 3 used/)
+    cy.get('[data-test="usage-contents"]').contains(/[0-500] of 500 used/)
```
Comment (herflis) [inline apps/sensenet/cypress/integration/dashboard/dashboard.spec.ts:26]:
```suggestion
    cy.get('[data-test="usage-contents"]').contains(/[0-9]|[1-8][0-9]|9[0-9]|[1-4][0-9]{2}|500 of 500 used/)
```

**Review by herflis — CHANGES_REQUESTED** [review]

File: apps/sensenet/cypress/integration/dashboard/dashboard.spec.ts (hunk 9 above)
Comment (NileshPatel17) [reply apps/sensenet/cypress/integration/dashboard/dashboard.spec.ts]: @herflis changes done

File: apps/sensenet/cypress/integration/dashboard/dashboard.spec.ts (hunk 10 above)
Comment (NileshPatel17) [reply apps/sensenet/cypress/integration/dashboard/dashboard.spec.ts]: @herflis changes done

**Comment by herflis** [comment]
@NileshPatel17 thank you for taking the time to help us! We greatly appreciate your contribution, you will get the extra sensenet-Hacktoberfest swag 🧦. We will contact you via email after the Hacktoberfest event. Hope you are having a great Hacktoberfest! 🎉 🙂

**Pull request merged**